code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -144,12 +151,43 @@
 			return 1
 		var/datum/computer_file/C = F.clone(0)
 		HDD.store_file(C)
+	if(href_list["PRG_encrypt"])
+		. = 1
+		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
+		if (!HDD || computer.enrolled != 2)
+			return
+		var/datum/computer_file/F = HDD.find_file_by_name(href_list["PRG_encrypt"])
+		if (!F)
+			return
+		F.password = sanitize(input(usr, "Enter an encryption key:", "Encrypt File"))
+	if(href_list["PRG_decrypt"])
+		. = 1
+		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
+		if (!HDD || computer.enrolled != 2)
+			return
+		var/datum/computer_file/F = HDD.find_file_by_name(href_list["PRG_encrypt"])
+		if (!F)
+			return
+		if (F.can_access_file(usr))
+			F.password = ""
+		else
+			return
 	if(.)
 		SSnanoui.update_uis(NM)
 
 /datum/nano_module/program/computer_filemanager
 	name = "NTOS File Manager"
 
+/datum/computer_file/proc/can_access_file(var/mob/user, input_password = "")
```

Comment:
This Proc should have been in `code/modules/modular_computers/file_system/computer_file.dm`